cvat/apps/lambda_manager/tests/test_lambda.py

```diff
@@ -966,3 +988,162 @@ def test_api_v2_lambda_functions_create_function_is_not_ready(self):
+                'name': 'test_task',
+                'labels': [{'name': 'cat'}],
+                'segment_size': 2
+            },
+            image_data=self._generate_task_images(6),
+            owner=self.admin,
+            org_id=self.org['id'],
+        )
+        self.task = task
+
+        jobs = self._get_request(f"/api/tasks/{self.task['id']}/jobs", self.admin,
+            org_id=self.org['id'])
+        assert jobs.status_code == status.HTTP_200_OK
+        self.job = jobs.json()[1]
+
+        self.common_data = {
+            "task": self.task['id'],
+            "frame": 0,
+            "cleanup": True,
+            "mapping": {
+                "car": { "name": "car" },
+            },
+        }
+
+        self.function_name = f"{LAMBDA_FUNCTIONS_PATH}/{id_function_detector}"
+
+        return super().setUp()
+
+    def _get_valid_job_params(self):
+        return {
+            "job": self.job['id'],
+            "frame": 2
+        }
+
+    def _get_invalid_job_params(self):
+        return {
+            "job": self.job['id'],
+            "frame": 0
+        }
+
+    def test_can_call_function_for_job_worker_in_org(self):
+        # not assigned + task => denied
+        # not assigned + job => denied
+        # task assignee + task => ok
+        # job assignee + task => denied
+        # job assignee + job => ok
+
+        data = self.common_data.copy()
+        with self.subTest(job=None, assignee=None):
+            response = self._post_request(self.function_name, self.user, data,
+                org_id=self.org['id'])
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+        data = self.common_data.copy()
+        data.update(self._get_valid_job_params())
+        with self.subTest(job='defined', assignee=None):
+            response = self._post_request(self.function_name, self.user, data,
+                org_id=self.org['id'])
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+        self._set_task_assignee(self.task['id'], self.user.id, org_id=self.org['id'])
+
+        data = self.common_data.copy()
+        with self.subTest(job=None, assignee='task'):
+            response = self._post_request(self.function_name, self.user, data,
+                org_id=self.org['id'])
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self._set_task_assignee(self.task['id'], None)
+
+
+        self._set_job_assignee(self.job['id'], self.user.id, org_id=self.org['id'])
+
+        data = self.common_data.copy()
+        with self.subTest(job=None, assignee='job'):
+            response = self._post_request(self.function_name, self.user, data,
+                org_id=self.org['id'])
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        self._set_job_assignee(self.job['id'], None)
+
+
+        self._set_job_assignee(self.job['id'], self.user.id, org_id=self.org['id'])
```

Comment:
No, the first one is for the cleanup, and the second one is a part of the next case. I'm still hoping to find a way to make these cases more separated. The overhead of splitting this test into separate smaller tests or introducing another derived class for these cases seems too big now, it can be done later.